csm/core/controllers/users.py

```diff
@@ -94,7 +94,7 @@ def __init__(self, request):
     async def check_max_user_limit(self):
         max_users_allowed = int(Conf.get(const.CSM_GLOBAL_INDEX, const.CSM_MAX_USERS_ALLOWED))
         existing_users_count = await self._service.get_user_count()
-        if existing_users_count >= max_users_allowed:
+        if existing_users_count >= max_users_allowed and max_users_allowed > 0:
```

Comment:
@avrnv 
In case of infinite csm users, max_users_allowed will be negative?

Reply:
Yes, sure. 0 or any negative number means infinite number of users.